pkg/v1/remote/transport/ping.go

```diff
@@ -39,22 +40,38 @@ const (
 // 300ms is the default fallback period for go's DNS dialer but we could make this configurable.
 var fallbackDelay = 300 * time.Millisecond
 
-type pingResp struct {
-	challenge challenge
+type PingResponse struct {
```

Comment:
I think we could get away with keeping these fields unexported, and have a separate unexported type with equivalent exported fields that we actually use to marshal. This jumps through some hoops, but it means our public API doesn't change any.